File: sdk/python/gptfrenzy_client.py
import requests, json, typing as _t


class GPTFrenzyClient:
    """Super-light client—no asyncio, no streaming."""

    def __init__(self, base_url: str = "http://localhost:8000"):
        self.base = base_url.rstrip("/")

    def chat(self, character: str, message: str) -> str:
        r = requests.post(
            f"{self.base}/chat",
            json={"character": character, "message": message},
            timeout=30,
        )
        r.raise_for_status()
        return r.json()["reply"]

    def manifest(self) -> _t.List[dict]:
        return requests.get(f"{self.base}/manifest", timeout=10).json()

File: sdk/csharp/GptFrenzyClient.cs
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using System.Collections.Generic;
using System.IO;
using Newtonsoft.Json;

public class GptFrenzyClient
{
    private readonly HttpClient _http;
    private readonly string _base;

    public GptFrenzyClient(string baseUrl = "http://localhost:8000")
    {
        _http = new HttpClient();
        _base = baseUrl.TrimEnd('/');
    }

    public async Task<string> Chat(string character, string message)
    {
        var payload = JsonConvert.SerializeObject(new { character, message });
        var resp = await _http.PostAsync($"{_base}/chat",
            new StringContent(payload, Encoding.UTF8, "application/json"));
        resp.EnsureSuccessStatusCode();
        dynamic body = JsonConvert.DeserializeObject(await resp.Content.ReadAsStringAsync());
        return (string)body.reply;
    }

    public async IAsyncEnumerable<string> ChatStream(string character, string message)
    {
        var payload = JsonConvert.SerializeObject(new { character, message });
        using var req = new HttpRequestMessage(HttpMethod.Post, $"{_base}/chat/stream")
        {
            Content = new StringContent(payload, Encoding.UTF8, "application/json")
        };
        var resp = await _http.SendAsync(req, HttpCompletionOption.ResponseHeadersRead);
        resp.EnsureSuccessStatusCode();
        var stream = await resp.Content.ReadAsStreamAsync();
        using var reader = new StreamReader(stream);
        while (true)
        {
            var line = await reader.ReadLineAsync();
            if (line == null)
                yield break;
            if (line.StartsWith("data: "))
                yield return line.Substring(6);
        }
    }

    public async Task<dynamic> Manifest()
    {
        var resp = await _http.GetAsync($"{_base}/manifest");
        resp.EnsureSuccessStatusCode();
        return JsonConvert.DeserializeObject(await resp.Content.ReadAsStringAsync());
    }
}
